File: 2° Periodo/Prog 2  JAVA C++/POO_TRABALHO_1/src/Paciente.java
public class Paciente extends Pessoa{
    private String relato;
    private String medicacaoConsumidaRegulamente;
    
    public Paciente(){
    }

    public Paciente(String nome, String endereco, String cpf, char sexo, int telefone, int identidade, String relato,String medicacaoConsumidaRegulamente) {
        super(nome, endereco, cpf, sexo, telefone, identidade);
        this.relato = relato;
        this.medicacaoConsumidaRegulamente = medicacaoConsumidaRegulamente;
    }

    public void imprimir(){
        super.imprimir();
        System.out.println("Relato: "+relato+"\nMedicação Consumida Regulamente: "+medicacaoConsumidaRegulamente);
    }

    
}

File: 2° Periodo/Prog 2  JAVA C++/LAB 10/src/Paciente.cpp
#include "Paciente.hpp"

Paciente::Paciente() {}

Paciente::Paciente(string nome, char sexo, string endereco, int cpf, string telefone, string identidade, string relato, string dataUltimaConsulta, string medicacaoQueToma)
    :Pessoa(nome,sexo,endereco,cpf,telefone, identidade){
        this->relato = relato;
        this->medicacaoQueToma = medicacaoQueToma;
        this->dataUltimaConsulta = dataUltimaConsulta;
}

void Paciente::imprimir() {
    Pessoa::imprimir();
    cout << "Relato: " << relato << endl;
    cout << "Data da Última Consulta: " << dataUltimaConsulta << endl;
    cout << "Medicação que Toma: " << medicacaoQueToma << endl;
}

void Paciente::setRelato(string relato) {
    this->relato = relato;
}

string Paciente::getRelato() {
    return relato;
}

void Paciente::setDataUltimaConsulta(string data) {
    dataUltimaConsulta = data;
}

string Paciente::getDataUltimaConsulta() {
    return dataUltimaConsulta;
}

void Paciente::setMedicacao(string medicacao) {
    medicacaoQueToma = medicacao;
}

string Paciente::getMedicacao() {
    return medicacaoQueToma;
}